javalib/src/main/scala/java/io/FileDescriptor.scala

```diff
@@ -1,43 +1,145 @@
 package java.io
 
-import scala.scalanative.unsigned._
-import scala.scalanative.unsafe._
+import java.io.FileDescriptor._
+import scala.scalanative.annotation.alwaysinline
+import scala.scalanative.meta.LinktimeInfo.isWindows
 import scala.scalanative.posix.{fcntl, unistd}
+import scala.scalanative.runtime.{Intrinsics, fromRawPtr}
+import scala.scalanative.unsafe._
+import scala.scalanative.unsigned._
+import scala.scalanative.windows.FileApi._
+import scala.scalanative.windows.FileApiExt._
+import scala.scalanative.windows.HandleApi._
+import scala.scalanative.windows.HandleApiExt._
+import scala.scalanative.windows.winnt.AccessRights._
+import scala.scalanative.windows.{ConsoleApiExt, DWord}
+
+final class FileDescriptor(fileHandle: FileHandle, readOnly: Boolean = false) {
+
+  def this() =
+    this {
+      if (isWindows) FileHandle(INVALID_HANDLE_VALUE)
+      else FileHandle(-1)
+    }
+
+  override def toString(): String =
+    s"FileDescriptor(${fd}, readonly=$readOnly"
+
+  /* Unix file descriptor underlying value */
+  @alwaysinline
+  private[java] def fd: Int = fileHandle.toInt
+
+  /* Windows file handler underlying value */
+  private[java] lazy val handle: Handle =
+    fromRawPtr[Byte](Intrinsics.castLongToRawPtr(fileHandle))
```

Comment:
Mainly to cache conversion to Ptr, however `def` with `@alwaysinline` annotation should fine, especially since it's going to be optimized anyway. I've replaced it with `def`